packages-exp/auth-exp/src/core/user/token_manager.test.ts

```diff
@@ -187,6 +193,137 @@ describe('core/user/token_manager', () => {
+        const updatedTokenManager = (authInternal.currentUser as UserImpl)
+          .stsTokenManager;
+
+        expect(signInRoute.calls[0].request).to.eql({
+          token: fakeCustomToken,
+          returnSecureToken: true
+        });
+        expect(tokens).to.eq('new-access-token');
+        expect(updatedTokenManager.accessToken).to.eq('new-access-token');
+        expect(updatedTokenManager.refreshToken).to.be.null;
+        expect(updatedTokenManager.expirationTime).to.eq(now + 3_600_000);
+        expect(updatedTokenManager.isPassthroughMode).to.be.true;
+      });
+
+      it('throws an error if forceRefresh is true and custom token provider is not set', async () => {
+        Object.assign(stsTokenManager, {
+          accessToken: 'token',
+          expirationTime: now + 100_000,
+          isPassthroughMode: true
+        });
+
+        await expect(stsTokenManager.getToken(auth, true)).to.be.rejectedWith(
+          FirebaseError,
+          'auth/internal-error'
+        );
+      });
+
+      it('returns access token during buffer window if custom token provider is not set', async () => {
+        Object.assign(stsTokenManager, {
+          accessToken: 'token',
+          expirationTime: now + Buffer.TOKEN_REFRESH,
+          isPassthroughMode: true
+        });
+
+        const tokens = (await stsTokenManager.getToken(auth))!;
+
+        expect(tokens).to.eql('token');
+      });
+
+      it('refreshes the token during buffer window if custom token provider is set', async () => {
+        Object.assign(stsTokenManager, {
+          accessToken: 'old-access-token',
+          expirationTime: now + Buffer.TOKEN_REFRESH / 2,
+          isPassthroughMode: true
+        });
+        setCustomTokenProvider(auth, provider);
+
+        const tokens = (await stsTokenManager.getToken(auth, true))!;
+        const authInternal = _castAuth(auth);
+        const updatedTokenManager = (authInternal.currentUser as UserImpl)
+          .stsTokenManager;
+
+        expect(signInRoute.calls[0].request).to.eql({
+          token: fakeCustomToken,
+          returnSecureToken: true
+        });
+        expect(tokens).to.eq('new-access-token');
+        expect(updatedTokenManager.accessToken).to.eq('new-access-token');
+        expect(updatedTokenManager.refreshToken).to.be.null;
+        expect(updatedTokenManager.expirationTime).to.eq(now + 3_600_000);
+        expect(updatedTokenManager.isPassthroughMode).to.be.true;
+      });
+
+      it('throws an error if the token is expired beyond the buffer window', async () => {
+        Object.assign(stsTokenManager, {
+          accessToken: 'old-access-token',
+          expirationTime: now - 1,
+          isPassthroughMode: true
+        });
+
+        await expect(stsTokenManager.getToken(auth)).to.be.rejectedWith(
+          FirebaseError,
+          "Firebase: The user's credential is no longer valid. The user must sign in again. (auth/user-token-expired)"
```

Comment:
nit: just use the error code substring here:
```ts
to.be.rejectedWith(FirebaseError, 'auth/user-token-expired')
```

Reply:
Ooh, got it, copied and pasted from another test in the file, will update that test too!